interface/admin.py

```diff
@@ -1,6 +1,46 @@
-from django.contrib import admin
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+from django.contrib import admin, messages
+from django.http import FileResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, queryset):
+        if queryset.count() != 1:
+            messages.warning(request, 'Only one assignment can be selected')
+            return request
```

Comment:
It seems weird to return a request, when the function returns responses, are you sure this is right? Maybe just `return`?

Reply:
It works either way. I'll leave `return` then.